detectors/node/opentelemetry-resource-detector-container/test/ContainerDetector.test.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as sinon from 'sinon';
+import * as assert from 'assert';
+import { Resource } from '@opentelemetry/resources';
+import { containerDetector } from '../src';
+import {
+  assertContainerResource,
+  assertEmptyResource,
+} from '@opentelemetry/contrib-test-utils';
+
+import { ContainerDetector } from '../src';
+
+describe('ContainerDetector', () => {
+  let readStub;
+  const correctCgroupV1Data =
+    'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklm';
+  const correctCgroupV2Data =
+    'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklm/hostname';
```

Comment:
Do you have any real life examples of full files? I think it would be worth to put them to test as well.